command/keyring/keyring.go

```diff
@@ -153,27 +172,41 @@ func (c *cmd) Run(args []string) int {
 	return 0
 }
 
-func (c *cmd) handleList(responses []*consulapi.KeyringResponse) {
-	for _, response := range responses {
-		pool := response.Datacenter + " (LAN)"
-		if response.Segment != "" {
-			pool += fmt.Sprintf(" [%s]", response.Segment)
-		}
-		if response.WAN {
-			pool = "WAN"
-		}
+func formatResponse(response *consulapi.KeyringResponse, keys map[string]int) string {
+	result := []string{
+		"",
+		poolName(response.Datacenter, response.WAN, response.Segment) + ":",
+		formatMessages(response.Messages),
+		formatKeys(keys, response.NumNodes),
+	}
+	return strings.Replace(strings.Join(result, "\n"), "\n\n", "\n", -1)
```

Comment:
Where does the `\n\n` come from? Potentially this could use `strings.Builder` as well, but I'm not sure if that removes the need for the `strings.Replace`.

Reply:
It was produced in case of eg messages where empty 

```golang
strings.Join([]string{"", "dc1", "", "key"}, "\n")
// "\ndc1\n\n"key"
```

but this doesn't happen anymore after my refactoring because I am using stringbuilder and relying on eg `formatMessages` to include its own newlines instead of `strings.Join` them.